lib/puppeteer/browser_runner.rb

```diff
@@ -161,14 +161,17 @@ def setup_connection(use_pipe:, timeout:, slow_mo:, preferred_revision:)
   end
 
   private def wait_for_ws_endpoint(browser_process, timeout, preferred_revision)
+    msg = []
```

Comment:
The name `msg` doesn't imply containing multiple items.
It would be better to name this with plural.

Reply:
I'd have to disagree with you here, sorry. I see `msg` as "the" (singular) contents of stderr, split into chunks. In this case, split into lines only because we happen match individual lines. Yes, they're collected in an array but I see this detail of choice of storage type as irrelevant to the naming of the variable -- it's just for memory efficiency, it shouldn't imply plurality of messages. It could just as easily have been a string, implying singularity per your logic, I assume, and the chunks would still have been appended to a singular `msg` the exact same way.

Sorry for the lengthy explanation, I hope it makes sense?

Reply:
Thank for the detailed explanation.

> I see this detail of choice of storage type as irrelevant to the naming of the variable -- it's just for memory efficiency, it shouldn't imply plurality of messages.

I got it. However I have a question. I understand `String#+` would cost a lot of memory, but   `String#<<` (concat) saves memory usage.
Is `Array#<<` and joining it better than `String#<<` ? (Sorry, I have few understanding of memory usage)

The reason I asked is that I thought  `String#<<` would be more representative of accumulating the chunks of message.